services/web/server/src/simcore_service_webserver/garbage_collector/_core_orphans.py

```diff
@@ -89,73 +89,72 @@ async def remove_orphaned_services(
+    potentially_running_service_ids: list[set[NodeID]] = []
+    async for project_nodes_future in limited_as_completed(
+        (
+            list_node_ids_in_project(app, project_id)
+            for project_id in known_opened_project_ids
+        ),
+        limit=_MAX_CONCURRENT_CALLS,
+    ):
+        try:
+            project_nodes = await project_nodes_future
+            potentially_running_service_ids.append(project_nodes)
+        except BaseException as e:  # pylint:disable=broad-exception-caught
+            _logger.warning(
+                create_troubleshotting_log_kwargs(
+                    (
+                        "Skipping orpahn services removal, call to "
+                        "`list_node_ids_in_project` raised"
                     ),
-                    exc_info=True,
-                )
-                return
-
-        potentially_running_service_ids_set: set[NodeID] = set().union(
-            *(node_id for node_id in potentially_running_service_ids)
-        )
-        _logger.debug(
-            "Allowed service UUIDs from known opened projects: %s",
-            potentially_running_service_ids_set,
-        )
-
-        # compute the difference to find the orphaned services
-        orphaned_running_service_ids = (
-            set(running_services_by_id) - potentially_running_service_ids_set
-        )
-        _logger.debug("Found orphaned services: %s", orphaned_running_service_ids)
-        # NOTE: no need to not reraise here, since we catch everything above
-        # and logged_gather first runs everything
-        await logged_gather(
-            *(
-                _remove_service(app, node_id, running_services_by_id[node_id])
-                for node_id in orphaned_running_service_ids
-            ),
-            log=_logger,
-            max_concurrency=_MAX_CONCURRENT_CALLS,
-        )
+                    error=e,
+                    error_context={
+                        "running_services": running_services,
+                        "running_services_by_id": running_services_by_id,
+                        "known_opened_project_ids": known_opened_project_ids,
+                    },
+                ),
+                exc_info=True,
+            )
+            continue
+
+    potentially_running_service_ids_set: set[NodeID] = set().union(
+        *(node_id for node_id in potentially_running_service_ids)
+    )
+    _logger.debug(
+        "Allowed service UUIDs from known opened projects: %s",
+        potentially_running_service_ids_set,
+    )
+
+    # compute the difference to find the orphaned services
+    orphaned_running_service_ids = (
+        set(running_services_by_id) - potentially_running_service_ids_set
+    )
+    _logger.debug("Found orphaned services: %s", orphaned_running_service_ids)
+    # NOTE: no need to not reraise here, since we catch everything above
+    # and logged_gather first runs everything
+    await logged_gather(
```

Comment:
this should also use limited_as_completed or limited_gather instead

Reply:
yup!